### ISSUE TITLE
Backups fail when using plugin outside of OADP/CAM context

### ISSUE THREAD

**Issue by dymurray** [openshift/openshift-velero-plugin#59]
Information to reproduce (from user):
1. We added the plugin openshift-velero-plugin and tried a Velero backup on a namespace with a pod, pvc and a service.
2. The backup seems to fail with the following message:
```
time="2020-10-06T19:33:45Z" level=info msg="[is-backup] Error in getting route: resource name may not be empty" backup=velero-ppdm/dk-k8s-12-2020-10-06-12-33-31-dk-k8s-12 cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/backup.go:48" pluginName=velero-plugins
time="2020-10-06T19:33:45Z" level=info msg="1 errors encountered backup up item" backup=velero-ppdm/dk-k8s-12-2020-10-06-12-33-31-dk-k8s-12 logSource="pkg/backup/backup.go:451" name=dk-k8s-12-service
```
3. Prior to this we were getting an error as shown below. We temporarily created an empty configMap “oadp-registry-config” In Velero namespace and landed with the above error in point 2. Do we mandatorily need to create a configMap in velero namespace.
```
time="2020-10-06T07:02:00Z" level=info msg="[is-backup] Error in getting route: configmaps \"oadp-registry-config\" not found" backup=velero-ppdm/test-os-2020-10-06-00-01-39-test-os cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/backup.go:48" pluginName=velero-plugins
time="2020-10-06T07:02:00Z" level=info msg="1 errors encountered backup up item" backup=velero-ppdm/test-os-2020-10-06-00-01-39-test-os
```

My thoughts:

I believe we introduced a regression when we merged the OADP work here. The intention was that if we didn't find the registry annotations we would simply assume we cannot backup the images in the cluster and we should just skip the IS plugins. We should also be documenting how a user can supply the needed information manually if they wish to include the image backups without using OADP/CAM.

**Comment by phuongatemc** [comment]
I hit this issue with OADP as well.
This is the error I got when I use OADP Backup:
```
time="2020-11-03T22:32:51Z" level=info msg="[GetRegistryInfo] value from imagestream" backup=oadp-operator/ph-backup cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/shared.go:35" pluginName=velero-plugins
time="2020-11-03T22:32:51Z" level=info msg="[is-backup] Error in getting route: configmaps \"oadp-registry-config\" not found" backup=oadp-operator/ph-backup cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/backup.go:48" pluginName=velero-plugins
time="2020-11-03T22:32:51Z" level=info msg="1 errors encountered backup up item" backup=oadp-operator/ph-backup logSource="pkg/backup/backup.go:451" name=mysql
time="2020-11-03T22:32:51Z" level=error msg="Error backing up item" backup=oadp-operator/ph-backup error="error executing custom action (groupResource=endpoints, namespace=mysql-ns, name=mysql): rpc error: code = Unknown desc = configmaps \"oadp-registry-config\" not found" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:455" name=mysql
time="2020-11-03T22:32:51Z" level=info msg="Backed up 19 items out of an estimated total of 30 (estimate will change throughout the backup)" backup=oadp-operator/ph-backup logSource="pkg/backup/backup.go:418" name=mysql namespace=mysql-ns progress= resource=endpoints
time="2020-11-03T22:32:51Z" level=info msg="Processing item" backup=oadp-operator/ph-backup logSource="pkg/backup/backup.go:378" name=mysql-ns namespace= progress= resource=namespaces
time="2020-11-03T22:32:51Z" level=info msg="Backing up item" backup=oadp-operator/ph-backup logSource="pkg/backup/item_backupper.go:121" name=mysql-ns namespace= resource=namespaces
time="2020-11-03T22:32:51Z" level=info msg="Executing custom action" backup=oadp-operator/ph-backup logSource="pkg/backup/item_backupper.go:327" name=mysql-ns namespace= resource=namespaces
time="2020-11-03T22:32:51Z" level=info msg="[common-backup] Entering common backup plugin" backup=oadp-operator/ph-backup cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/backup.go:25" pluginName=velero-plugins
time="2020-11-03T22:32:51Z" level=info msg="[GetRegistryInfo] value from imagestream" backup=oadp-operator/ph-backup cmd=/plugins/velero-plugins logSource="/go/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/common/shared.go:35" pluginName=velero-plugins
...

And here is the yaml of my Backup object:
apiVersion: velero.io/v1
kind: Backup
metadata:
  annotations:
    velero.io/source-cluster-k8s-gitversion: v1.17.1
    velero.io/source-cluster-k8s-major-version: '1'
    velero.io/source-cluster-k8s-minor-version: 17+
  selfLink: /apis/velero.io/v1/namespaces/oadp-operator/backups/ph-backup
  resourceVersion: '20351045'
  name: ph-backup
  uid: 81c8ef1e-ff12-4af2-bfdd-fa4da0537c2d
  creationTimestamp: '2020-11-03T22:32:32Z'
  generation: 9
  namespace: oadp-operator
  labels:
    velero.io/storage-location: plc-mysql-app-vm-13-13-fa3cf.stagings
spec:
  defaultVolumesToRestic: false
  excludedResources:
    - persistentvolumeclaims
    - persistentvolumes
  includedNamespaces:
    - mysql-ns
  storageLocation: plc-mysql-app-vm-13-13-fa3cf.stagings
  ttl: 720h0m0s
status:
  completionTimestamp: '2020-11-03T22:32:53Z'
  errors: 30
  expiration: '2020-12-03T22:32:33Z'
  formatVersion: 1.1.0
  phase: PartiallyFailed
  progress:
    itemsBackedUp: 30
    totalItems: 30
  startTimestamp: '2020-11-03T22:32:33Z'
  version: 1

**Comment by phuongatemc** [comment]
This is the Velero yaml on OADP
--------------------------------------
apiVersion: konveyor.openshift.io/v1alpha1
kind: Velero
metadata:
  creationTimestamp: '2020-10-30T15:39:10Z'
  generation: 1
  name: example-velero
  namespace: oadp-operator
  resourceVersion: '18007776'
  selfLink: >-
    /apis/konveyor.openshift.io/v1alpha1/namespaces/oadp-operator/veleros/example-velero
  uid: 567ade07-9bf7-4299-8588-26fd86b4db15
spec:
  backup_storage_locations:
    - config: null
      credentials_secret_ref:
        name: ph-test-plc-vm-13-13-0c2ee
        namespace: powerprotect
      ddHost: 10.235.12.193
      name: ph-test-plc-vm-13-13-0c2ee
      namespace: powerprotect
      object_storage:
        bucket: ph-test-plc-vm-13-13-0c2ee
        prefix: Backups/d9dbd422-6cf6-4007-9008-33e114cf3980/velerobackups
      prefix: Backups/d9dbd422-6cf6-4007-9008-33e114cf3980/velerobackups
      provider: ppdm.dellemc.com/velero-dd-plugin
  custom_velero_plugins:
    - image: 'ap-sc.lss.emc.com:8446/dellemc/powerprotect-velero-dd:19.6.0-1-SNAPSHOT'
      name: velero-dd-plugin
  default_velero_plugins:
    - csi
    - openshift
  enable_restic: true
  olm_managed: true
  use_upstream_images: true
status:
  conditions:
    - ansibleResult:
        changed: 0
        completion: '2020-10-30T15:40:25.883609'
        failures: 0
        ok: 7
        skipped: 58
      lastTransitionTime: '2020-10-30T15:39:10Z'
      message: Awaiting next reconciliation
      reason: Successful
      status: 'True'
      type: Running

**Comment by phuongatemc** [comment]
I'm using internal copy of Data Domain plugin, but you can access the same image at https://hub.docker.com/r/dellemc/powerprotect-velero-dd/tags
or simply specify "image: dellemc/powerprotect-velero-dd:19.6.0-3"

**Comment by dymurray** [comment]
@phuongatemc I see an invalid configuration in velero yaml:
```
backup_storage_locations:
- config: null
```


OADP cannot work properly with this configuration.

This github issue is related to the fact that this plugin fails outside of OADP context. I will get that PR rebased and merged to fix that issue but I'm afraid it doesn't relate to the issue you are seeing using OADP. You need to fix that configuration of the BSL.

**Comment by dymurray** [comment]
Fixed by #61
